tests/test_repository.py

```diff
@@ -186,3 +186,26 @@ def test_sle15_packages(container_per_test, pkg):
     remain installable and available.
     """
     container_per_test.connection.run_expect([0], f"zypper -n in {pkg}")
+
+
+@pytest.mark.parametrize("container_per_test", [BASE_CONTAINER], indirect=True)
+def test_orphaned_packages(container_per_test):
+    """Test installed packages in the container are also available in the BCI repository
+    and that there are no orphanes left, except few special cases
+    """
+    EXPECTED_ORPHANS = [
+        "kubic-locale-archive",
+        "skelcd-EULA-bci",
+        "sles-release",
+    ]
+
+    raw = container_per_test.connection.check_output(
+        "zypper -q -s 11 pa --orphaned"
+    )
+    tokens = raw.split()
+    detected_orphans = [
+        tokens[i + 1] for i, v in enumerate(tokens) if v == "@System"
+    ]
+
+    for pkg in detected_orphans:
+        assert pkg in EXPECTED_ORPHANS
```

Comment:
Longer, but imho clearer.

```suggestion
    lines = container_per_test.connection.check_output(
        "zypper -q -s 11 pa --orphaned"
    ).splitlines()

    # output of zypper -q -s 11 pa --orphaned:
    # S   Repository  Name                  Version                Arch
    #
    # i+  @System     kubic-locale-archive  2.31-10.36             noarch
    orphans = []
    for line in lines:
        columns = line.split()
        if len(columns) == 5:
            _, location, orphan_name, _, _ = columns
            if location == "@System":
                orphans.append(orphan_name)

    assert set(orphans) == set(EXPECTED_ORPHANS)
```

Reply:
This is nice, thanks. Should we expect there will be always the same set of expected orphans. I thought that if any of those would disappear in future (for instance `kubic-locale-archive`) it should not be a problem.

Reply:
I don't know whether the orphans will stay the same. I'd just go ahead and hardcode them for now. If it turns into a problem, we can always adjust the tests later on.